### ISSUE TITLE
[question] Propagate a variable to dependency package 

### ISSUE THREAD

**Issue by vithalsm** [conan-io/conan#14406]
### What is your question?

Have a conan package, p1 that dependent on another conan package p2.   Dependency is defined using requires() i
>>p1/conanfile.py defines
    def requirements()
      <set-and-export-a-variable>, say ptype="c"
       requires("p2").

Now, p1 wants to define a variable, and export it to p2.  p2/conanfile.py needs to access the same under 'requirement()' method.
>> p2/conanfile.py
     def requirements()
         <get-and-ref-the-variable-here>
         if (ptype=="c") 
            requires("xyz")
         else
            requires("abc")

As shown above, dependent recipe defines 'requirements' based on a variable set by the consumer.
Please let me know how to achieve this.  Whether through a variable as  shown above or any other methods that conan provides.

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @vithalsm 

It is not very clear what you mean with ``Now, p1 wants to define a variable, and export it to p2``

I guess that ``options`` is what you want. Please check 
- https://docs.conan.io/2/tutorial/creating_packages/configure_options_settings.html 
- https://docs.conan.io/2/reference/conanfile/attributes.html#options
- https://docs.conan.io/2/reference/conanfile/attributes.html#default-options

Also, options can be defined in profiles (better than defined by downstream consumers): 
- https://docs.conan.io/2/reference/config_files/profiles.html#options

**Comment by vithalsm** [comment]
Profile - Is not the options for my case.
Options - Can the options set by the consumer package (p1) be accessed within the dependent package (p2)?  If so, yes, that meets my requirement.  Please confirm.

**Comment by memsharded** [comment]
Yes, every package can read its options with ``self.options.myoption``, and consumers can define ``default_options = {"mydep/*:myoption": "myvalue"}`` to try to assign it. Note that multiple packages defining different values for the same dependency can happen, and only the first value is taken into account. This is the reason why defining options in profiles is better, as it is more global, not subject to differences between packages.

**Comment by vithalsm** [comment]
Nice.  Thank you @memsharded. I shall try that.

**Comment by AbrilRBS** [comment]
Closing as solved as we didn't hear back from you in a while, we hope that the proposed solution worked, thanks!
